core-rust/mesh-api-server/src/mesh_api/handlers/construction_parse.rs

```diff
@@ -53,7 +49,7 @@ pub(crate) async fn handle_construction_parse(
 
     let mapping_context = MappingContext::new(&state.network);
     let database = state.state_manager.database.snapshot();
```

Comment:
I've just noticed that we still say `Invalid transaction` if someone tries to parse some other transaction such as a V2 transaction - I wonder if we can throw a bit of a nicer error in this case - e.g.
```rust
        let transaction = match manifest_decode::<AnyTransaction>(&transaction_bytes) {
            Ok(AnyTransaction::NotarizedTransactionV1(transaction)) => transaction,
            Ok(_) => {
                return Err(ResponseError::from(ApiError::InvalidTransaction)
                        .with_details("Only V1 notarized transactions are supported in the Mesh API parse endpoint"));
            },
            Err(_) => {
                return Err(ResponseError::from(ApiError::InvalidTransaction)
                        .with_details("Invalid transaction"));
            }
        };
```

Reply:
Done